src/tools/schema2packager.ts

```diff
@@ -219,24 +225,58 @@ struct std::hash<arcs::${name}> {
     return std::hash<std::string>()(entity._internal_id());
   }
 };
-
-#endif
 `;
   return content.replace(/ +\n/g, '\n');
 }
 
-// TODO: handle schemas with multiple names and schemas with parents
-// TODO: error handling
-async function processFile(file) {
-  const contents = fs.readFileSync(file, 'utf-8');
+// TODO: schemas with multiple names and schemas with parents
+// TODO: schemas with no fields
+// TODO: schemas with no names?  i.e. inline '* {Text s}'
+async function processFile(src) {
+  const outName = path.basename(src).toLowerCase().replace(/\./g, '-') + '.h';
+  const outPath = path.join(opts.outdir, outName);
+  console.log(outPath);
+  if (opts.update && fs.existsSync(outPath) && fs.statSync(outPath).mtimeMs > fs.statSync(src).mtimeMs) {
+    return;
+  }
+
+  const contents = fs.readFileSync(src, 'utf-8');
   const manifest = await Manifest.parse(contents);
-  for (const schema of Object.values(manifest.schemas)) {
-    const outFile = 'entity-' + schema.names[0].toLowerCase() + '.h';
-    const contents = generate(schema.names[0], schema);
-    fs.writeFileSync(outFile, contents);
+
+  // Collect declared schemas along with any inlined in particle connections.
+  const schemas: Dictionary<Schema> = {...manifest.schemas};
+  for (const particle of manifest.particles) {
+    for (const connection of particle.connections) {
+      const schema = connection.type.getEntitySchema();
```

Comment:
Is this for "in-line" (for lack of a better term) schemas? Or schemas that are defined within particles?

Reply:
Yes, this catches inline schemas.